cocos/physics/cannon/cannon-world.ts

```diff
@@ -203,6 +249,52 @@ export class CannonWorld implements IPhysicsWorld {
             this._world.removeConstraint(constraint.impl);
         }
     }
+
+    get debugDrawFlags (): EPhysicsDrawFlags {
+        return this._debugDrawFlags;
+    }
+
+    set debugDrawFlags (v: EPhysicsDrawFlags) {
+        this._debugDrawFlags = v;
+    }
+
+    get debugDrawConstraintSize (): number {
+        return this._debugConstraintSize;
+    }
+
+    set debugDrawConstraintSize (v) {
+        this._debugConstraintSize = v;
+    }
+
+    private _getDebugRenderer (): GeometryRenderer|null {
+        const cameras = director.root!.mainWindow?.cameras;
+        if (!cameras) return null;
+        if (cameras.length === 0) return null;
+        if (!cameras[0]) return null;
+        cameras[0].initGeometryRenderer();
+
+        return cameras[0].geometryRenderer;
+    }
+
+    private _debugDraw (): void {
+        const debugRenderer = this._getDebugRenderer();
+        if (!debugRenderer) return;
+
+        this._debugLineCount = 0;
+        if (this._debugDrawFlags & EPhysicsDrawFlags.AABB) {
+            for (let i = 0; i < this.bodies.length; i++) {
+                const body = this.bodies[i];
+                for (let j = 0; j < body.wrappedShapes.length; j++) {
+                    const shape = body.wrappedShapes[j];
+                    if (this._debugLineCount + 12 < this._MAX_DEBUG_LINE_COUNT) {
+                        this._debugLineCount += 12;
```

Comment:
What does 12 means?

Reply:
> What does 12 means?

12 lines of a AABB

Reply:
It's best to define a constant to represent it